pantheon-bundle/src/main/java/com/redhat/pantheon/servlet/PublishDraftVersion.java

```diff
@@ -80,16 +81,21 @@ public void run(SlingHttpServletRequest request, PostResponse response, SlingPos
         if (response.getError() == null) {
             // call the extension point
             Locale locale = getLocale(request);
-            Module module = getModule(request);
+            Optional<? extends Document> document = getDocument(request, request.getResourceResolver());
             String variant = getVariant(request);
-            ModuleVersion moduleVersion = module.locale(locale).get()
+            Optional<? extends DocumentVersion> documentVersion = Optional.of(document.get().locale(locale).get()
                     .variants().get()
                     .variant(variant).get()
-                    .released().get();
+                    .released().get());
+
+            // Regenerate the document once more
+            asciidoctorService.getModuleHtml(document.get(), locale, variant, false, new HashMap(), true);
 
-            // Regenerate the module once more
-            asciidoctorService.getModuleHtml(module, locale, variant, false, new HashMap(), true);
-            events.fireEvent(new ModuleVersionPublishedEvent(moduleVersion), 15);
+            if(documentVersion.get() instanceof AssemblyVersion){
+                events.fireEvent(new AssemblyVersionPublishedEvent((AssemblyVersion) documentVersion.get()), 15);
+            }else{
+                events.fireEvent(new ModuleVersionPublishedEvent((ModuleVersion)documentVersion.get()), 15);
+            }
```

Comment:
This is the only place where you need to know whether it's assembly vs module. I would not use `instanceof` here - rather, I would just use `if ("pantheon/assemblyVersion".equals(documentVersion.getResourceType())` or something similar. Let me know if you don't understand why, we can talk about the difference on chat.